core/src/test/scala/unit/kafka/tools/MirrorMakerTest.scala

```diff
@@ -17,12 +17,82 @@
 
 package kafka.tools
 
-import kafka.consumer.BaseConsumerRecord
+import java.util
+import java.util.Properties
+
+import kafka.consumer.{BaseConsumerRecord}
+import kafka.server.{KafkaConfig, KafkaServer}
+import kafka.tools.MirrorMaker.MirrorMakerNewConsumer
+import kafka.utils.{CoreUtils, TestUtils}
+import kafka.zk.ZooKeeperTestHarness
+import org.apache.kafka.clients.consumer.{ConsumerConfig, KafkaConsumer}
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
 import org.apache.kafka.common.record.{Record, TimestampType}
 import org.junit.Assert._
-import org.junit.Test
+import org.junit.{After, Before, Test}
+
+class MirrorMakerTest extends ZooKeeperTestHarness {
+
+  private var server1: KafkaServer = null
+
+  @Before
+  override def setUp() {
+    super.setUp()
+    val props1 = TestUtils.createBrokerConfig(0, zkConnect, false)
+    val config1 = KafkaConfig.fromProps(props1)
+    server1 = TestUtils.createServer(config1)
 
-class MirrorMakerTest {
+    val topic = "new-topic"
+    TestUtils.createTopic(zkUtils, topic, numPartitions = 1, replicationFactor = 1, servers = List(server1))
+  }
+
+  @After
+  override def tearDown() {
+    server1.shutdown
+    CoreUtils.delete(server1.config.logDirs)
+    super.tearDown()
+  }
+
+  @Test
+  def testWhitelistTopic() {
+    val topic = "new-topic"
+    val msg = "test message"
+    val brokerList = TestUtils.getBrokerListStrFromServers(Seq(server1))
+
+    // Create a test producer to delivery a message
+    val props = new Properties()
+    props.put("bootstrap.servers", brokerList)
+    props.put("key.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+    props.put("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+    val producer1 = new KafkaProducer[Array[Byte], Array[Byte]](props)
+    producer1.send(new ProducerRecord[Array[Byte], Array[Byte]](topic, msg.getBytes()))
+    producer1.flush() // Explicitly invoke flush method to make effect immediately
+    producer1.close() // Close the producer
+
+    // Create a MirrorMaker consumer
+    val config = new util.HashMap[String, AnyRef]
+    config.put(ConsumerConfig.GROUP_ID_CONFIG, "test-gropu")
+    config.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    config.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+    config.put("key.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer")
+    config.put("value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer")
+    val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](config)
+    MirrorMaker.createMirrorMakerProducer(brokerList)
+
+    val whitelist = Some("new.*|another_topic|foo" ) // Test a regular expression
```

Comment:
Yeah, that should fail without your fix, I think. If we wanted to make the test even better, we'd produce to `another_topic` and `yet_another_topic` and check that the consumer only consumes from the two that match.

But the test as it is verifies the fix, so I won't block the merge if you don't feel like extending the test. :)